pkg/scale/btree.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package scale
+
+import (
+	"reflect"
+
+	"github.com/tidwall/btree"
+)
+
+// BTree is a wrapper around tidwall/btree.BTree that also stores the comparator function and the type of the items
+// stored in the BTree. This is needed during decoding because the BTree is a generic type, and we need to know the
+// type of the items stored in the BTree in order to decode them.
+type BTree struct {
+	*btree.BTree
+	Comparator func(a, b interface{}) bool
```

Comment:
I'm curious why BTree is storing Comparator?  I see needing it to create the new BTree, but I'm not sure I understand why it's  in the BTree struct.

Reply:
You're right. We need it to decode a BTree. 

Since we are encoding just the data in BTree and not the comparator, we'll need a reference to it during the decoding in order to create a new `btree.BTree`. 

The flow for decoding would be
- Create a new instance of `scale.BTree` by passing the type of the objects in the `btree.BTree` and the comparator method
- Call `scale.Unmarshal` with the initialised `BTree` struct

I couldn't think of any other way of handling this and I am open to suggestions!